### PR TITLE
Apigee install script in Python

### PR DESCRIPTION
**Pull request by michmerr**
### Description of the Change
Following changes are necessary to prepare the Apigee Platform to be able to support Jazz serverless framework. init script deploys the required Common API Proxy as well as create other configuration dependencies within the Apigee API Platfrom.

### Benefits
This change will create an interface between the Apigee and AWS ecosystems for Apigee to be able to serve the API requests on behalf of the serverless functions created through Jazz.

### Possible Drawbacks
None as this change is independent and limited to Apigee API Platform.

### Applicable Issues
Need to update the templates for the OAuth functionality

### CODE REVIEW

File: feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/DataEncryption.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (bleggett) [inline feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/DataEncryption.xml]: Speaking of zipfiles, can't you bundle all the shared flow stuff into a zip and deploy it as one unit?

I'm asking, I'm not sure of what the Apigee Way is.

**Comment by bleggett** [comment]
@michmerr Can you correct the lint errors that are causing the Travis checks to fail on your Python script? You can install [flake8](http://flake8.pycqa.org/en/latest/) and run it against your script manually, or if your editor supports it (many do) you can see the issues pop up while editing.

Also:

-  feature-extensions/apigee/apigee-installer/Common-Jazz.zip: We are trying to avoid putting binary blobs into this repo, can you raise a PR against `tmobile/jazz-content` and put that file there? You should be able to `curl` the file from that github repo to the local disk at install time in your scripts.

- Thanks for adding some unit tests!

- feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/resources/java/AESEncryption.jar: Do we have to include this `jar` in source control? Can it be installed or downloaded at runtime with some sort of dependency management tool (maven or gradle or whatever, I don't keep up with the Java ecosystem)? If not, it should also go into `jazz-content`

**Comment by devsatishm** [comment]
@michmerr @svidhani Added you to jazz-content repo.

File: feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/DataEncryption.xml (hunk 1 above)
Comment (muditpurwar) [reply feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/DataEncryption.xml]: Each sharedflow is a deployable unit and hence cannot be bundled into one zip.

They are generic sharedflows and some of them are customized for Jazz use.

**Comment by muditpurwar** [comment]
> @michmerr Can you correct the lint errors that are causing the Travis checks to fail on your Python script? You can install [flake8](http://flake8.pycqa.org/en/latest/) and run it against your script manually from the repo root, or if your editor supports it (many do with extensions) you can see the issues pop up while editing.
> 
> Also:
> 
> * `feature-extensions/apigee/apigee-installer/Common-Jazz.zip`: We are trying to avoid putting binary blobs into this repo, can you raise a PR against `tmobile/jazz-content` and put that file there? You should be able to `curl` the file from that github repo to the local disk at install time in your scripts.

- Yes, its ok to move the Common-Jazz.zip to a different repo - tmobile/jazz-content and download it at run time.

> * Thanks for adding some unit tests!
> * `feature-extensions/apigee/apigee-installer/sharedflows/DataEncryption/sharedflowbundle/resources/java/AESEncryption.jar`: Do we have to include this `jar` in source control? Can it be installed or downloaded at runtime with some sort of dependency management tool (maven or gradle or whatever, I don't keep up with the Java ecosystem), or even just curl? If not, it should also go into `jazz-content`

- No, this cannot be moved as it has to be part of the sharedflow installer as a resource.

**Comment by bleggett** [comment]
@michmerr Thanks! @suryajak @devsatishm Do you guys want to put `Common_Jazz.zip` into `jazz-content` or are you OK with leaving it here?

Other than that LGTM.

**Comment by muditpurwar** [comment]
> @michmerr Thanks! @suryajak @devsatishm Do you guys want to put `Common_Jazz.zip` into `jazz-content` or are you OK with leaving it here?
> 
> Other than that LGTM.

@bleggett -We can leave it in this repo only. It will be a single repo for all install dependencies.

**Comment by suryajak** [comment]
@muditpurwar : We want all binary files to go into `jazz_content` repo. Is there a reason `Common_Jazz.zip` cannot go to `jazz_content`?

**Comment by muditpurwar** [comment]
> @muditpurwar : We want all binary files to go into `jazz_content` repo. Is there a reason `Common_Jazz.zip` cannot go to `jazz_content`?

We are ok to move it to the 'jazz_content' repo, we just need to update the install script to download at runtime.

**Comment by michmerr** [comment]
Moved Common-Jazz.zip to jazz-content repo.

(see https://github.com/tmobile/jazz-content/pull/5)

Leaving AESEncryption.jar in jazz-installer per @muditpurwar's comment.

**Comment by bleggett** [comment]
Thanks @muditpurwar @michmerr !

**Pull request merged**